src/shared/sleep-util.c

```diff
@@ -1028,6 +1028,45 @@ int read_fiemap(int fd, struct fiemap **ret) {
         return 0;
 }
 
+int write_resume_config(dev_t devno, uint64_t offset, const char *device) {
+        char offset_str[DECIMAL_STR_MAX(uint64_t)];
+        _cleanup_free_ char *path = NULL;
+        const char *devno_str;
+        int r;
+
+        devno_str = FORMAT_DEVNUM(devno);
+        xsprintf(offset_str, "%" PRIu64, offset);
+
+        if (!device) {
+                r = device_path_make_canonical(S_IFBLK, devno, &path);
+                if (r < 0)
+                        return log_oom();
```

Comment:
device_path_make_canonical() can actually return all kinds of errors, maybe use `log_error_errno(r, "Failed to format canonical device path for device %s: %m")`